src/openrct2/network/NetworkBase.cpp

```diff
@@ -715,11 +715,65 @@ const char* NetworkBase::FormatChat(NetworkPlayer* fromplayer, const char* text)
 {
     static std::string formatted;
     formatted.clear();
-    formatted += "{OUTLINE}";
+
     if (fromplayer != nullptr)
     {
-        formatted += "{BABYBLUE}";
-        formatted += fromplayer->Name;
+        auto& network = OpenRCT2::GetContext()->GetNetwork();
+        auto it = network.GetGroupByID(fromplayer->Id);
+        std::string groupname = "";
+        std::vector<std::string> colors;
+        if (it != nullptr)
+        {
+            groupname = it->GetName();
+            if (groupname[0] != '{')
+            {
+                colors.push_back("{WHITE}");
+            }
+        }
+
+        for (size_t i = 0; i < groupname.size(); ++i)
+        {
+            if (groupname[i] == '{')
+            {
+                std::string color = "{";
+                ++i;
+                while (i < groupname.size() && groupname[i] != '}' && groupname[i] != '{')
+                {
+                    color += groupname[i];
+                    ++i;
+                }
+                color += '}';
+                if (groupname[i] == '}' && i < groupname.size())
+                {
+                    colors.push_back(color);
+                }
+            }
+        }
+
+        if (colors.size() == 0 || (colors.size() == 1 && colors[0] == "{WHITE}"))
+        {
+            formatted += "{BABYBLUE}";
+            formatted += fromplayer->Name;
+        }
+        else
+        {
+            size_t j = 0;
+            for (size_t i = 0; i < colors.size(); ++i)
+            {
+                formatted += colors[i];
+                size_t numcharacters = (fromplayer->Name.size() / colors.size()) + j;
```

Comment:
I don't expect player name or colors to change in each iteration, please move `fromplayer->Name.size() / colors.size()` calculation to before the loop.

Reply:
Moved!